MeshGeoToolsLib/GeoMapper.cpp

```diff
@@ -67,25 +77,46 @@ void GeoMapper::mapOnMesh(const std::string &file_name)
 
 void GeoMapper::mapOnMesh(const MeshLib::Mesh* mesh)
 {
+	std::vector<GeoLib::Point*> const* pnts(_geo_objects.getPointVec(_geo_name));
+	if (! pnts) {
+		ERR("Geometry \"%s\" does not exist.", _geo_name.c_str());
+		return;
+	}
+
+	// the variable _surface_mesh is reused below, so first the existing
+	// _surface_mesh has to be cleaned up
+	if (_surface_mesh)
+		delete _surface_mesh;
+
 	if (mesh->getDimension()<3)
-		this->_surface_mesh = new MeshLib::Mesh(*mesh);
+		_surface_mesh = new MeshLib::Mesh(*mesh);
 	else
 	{
 		const MathLib::Vector3 dir(0,0,-1);
-		this->_surface_mesh = MeshLib::MeshSurfaceExtraction::getMeshSurface(*mesh, dir, 90);
+		_surface_mesh = MeshLib::MeshSurfaceExtraction::getMeshSurface(*mesh, dir, 90);
 	}
 
 	// init grid
 	MathLib::Point3d origin(std::array<double,3>{{0,0,0}});
 	MathLib::Vector3 normal(0,0,-1);
-	MeshLib::Mesh const*const flat_mesh =
-		MeshLib::projectMeshOntoPlane(*_surface_mesh, origin, normal);
-	std::vector<MeshLib::Node*> const& flat_nodes (flat_mesh->getNodes());
+	std::vector<MeshLib::Node*> flat_nodes;
```

Comment:
This could be simple vector of Nodes, because you make a copy + change it later.
```diff
diff --git a/MeshGeoToolsLib/GeoMapper.cpp b/MeshGeoToolsLib/GeoMapper.cpp
index 39cd19d..d46284d 100644
--- a/MeshGeoToolsLib/GeoMapper.cpp
+++ b/MeshGeoToolsLib/GeoMapper.cpp
@@ -99,12 +99,13 @@ void GeoMapper::mapOnMesh(const MeshLib::Mesh* mesh)
        // init grid
        MathLib::Point3d origin(std::array<double,3>{{0,0,0}});
        MathLib::Vector3 normal(0,0,-1);
-       std::vector<MeshLib::Node*> flat_nodes;
+       std::vector<MeshLib::Node> flat_nodes;
        // copy nodes and project the copied nodes to the x-y-plane, i.e. set
        // z-coordinate to zero
+       flat_nodes.reserve(_surface_mesh->getNNodes());
        for (auto n_ptr : _surface_mesh->getNodes()) {
-               flat_nodes.push_back(new MeshLib::Node(*n_ptr));
-               (*flat_nodes.back())[2] = 0.0;
+               flat_nodes.emplace_back(*n_ptr);
+               flat_nodes.back()[2] = 0.0;
        }
        _grid = new GeoLib::Grid<MeshLib::Node>(flat_nodes.cbegin(), flat_nodes.cend());
 
@@ -115,8 +116,6 @@ void GeoMapper::mapOnMesh(const MeshLib::Mesh* mesh)
        }
 
        delete _grid;
-       for (auto n_ptr : flat_nodes)
-               delete n_ptr;
 }
 
 void GeoMapper::mapToConstantValue(double value)
```